apps/backend/src/users/users.service.ts

```diff
@@ -26,11 +29,19 @@ export class UsersService {
     return user;
   }
 
-  update(id: string) {
-    return `This action updates a #${id} user`;
+  async update(id: string, data: UpdateUserDto): Promise<User> {
+    try {
+      return await this.prisma.user.update({ where: { authSchId: id }, data });
+    } catch {
+      throw new NotFoundException(`User not found`);
```

Comment:
Is this the only error we can get here? Can we get an error if the user is found, but the data s not correct?

Reply:
I believe that only realistic error we can get is NotFoundException, since the point of issue #15 was to validate all the data before it reached Prisma. Isn't that correct?

Reply:
That's true, but can you please try the endpoints with Postmen to see if such errors are possible? And if not then it is good like this.

Reply:
I think the email is unique, so it can give a unique constraint violation. Btw we shouldn't allow the user to change their name, it should be the same as the one from autsch.